packages/nuxt/src/runtime/composables.ts

```diff
@@ -1,4 +1,27 @@
 import { useNuxtApp } from '#app'
+import {
+  defineStore as _defineStore,
+  type Pinia,
+  type StoreGeneric,
+} from 'pinia'
 export * from 'pinia'
 
-export const usePinia = () => useNuxtApp().$pinia
+export const usePinia = () => useNuxtApp().$pinia as Pinia | undefined
+
+export const defineStore: typeof _defineStore = (
```

Comment:
Let's make the error dev only, it shouldn't affect the final bundle. This should be possible by using `process.env.NODE_ENV === 'production' ? _defineStore : ...` or a similar guard